Boltdir/site-modules/gplt/tasks/create_control_repo_from_production_env.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+ENVIRONMENT=production
+PUPPET_PROD_ENV_DIR=/etc/puppetlabs/code/environments/$ENVIRONMENT
+BASEDIR=/opt/puppet
+CONTROL_REPO_URL=$BASEDIR/control-repo.git
+CHECKOUT_DIR=/root/control-repo
+
+mkdir -p $BASEDIR
+
+if [ -d $CONTROL_REPO_URL ]; then
+    echo "ERROR: The control repo destination already exists" >&2
+    exit 1
+fi
+
+git init --bare $CONTROL_REPO_URL
+git clone $CONTROL_REPO_URL $CHECKOUT_DIR
+cp -r $PUPPET_PROD_ENV_DIR/* $CHECKOUT_DIR
+rm -fr $CHECKOUT_DIR/modules
+cd $CHECKOUT_DIR \
+    || { echo "ERROR: Failed to cd into $CHECKOUT_DIR" >&2 ; exit 1 ; }
+git add .
+git commit -m "Initial commit"
+git branch -m $ENVIRONMENT
+git push -u origin $ENVIRONMENT
```

Comment:
Do we need to do any checking about these git commands exist status?  Like if somehow it failed at the end, would bolt give us a useful error message, or would it expect us to produce something?

Reply:
No, if any of the above fail, the script will exit because we are not ignoring failures. The contents of STDERR are then echoed via bolt as part of the failure json response.